File: android/app/src/main/java/com/rbdarts/feature/stats/StatsService.kt
package com.rbdarts.feature.stats

import com.rbdarts.core.domain.GameSummary
import com.rbdarts.core.domain.ParticipantScorecard
import com.rbdarts.core.domain.ScoringRules
import com.rbdarts.core.domain.TeamStanding

class StatsService {
    fun gameSummary(participants: List<ParticipantScorecard>): GameSummary =
        ScoringRules.summarizeStandalone(participants)

    fun standings(standings: List<TeamStanding>): List<TeamStanding> =
        ScoringRules.rankedStandings(standings)
}

File: ios/RBDarts/Features/Stats/StatsService.swift
import Foundation

public struct StatsService: Sendable {
    public init() {}

    public func gameSummary(participants: [ParticipantScorecard]) -> GameSummary {
        ScoringRules.summarizeStandalone(participants: participants)
    }

    public func standings(_ standings: [TeamStanding]) -> [TeamStanding] {
        ScoringRules.rankedStandings(standings)
    }
}
